presto-docs/src/main/sphinx/connector/elasticsearch.rst

```diff
@@ -0,0 +1,171 @@
+Property Name                           Description
+======================================= ==============================================================================
+``elasticsearch.table-names``           List of tables provided by the catalog
+``elasticsearch.default-schema``        Default schema name for tables
+``elasticsearch.table-description-dir`` Directory containing JSON table description files
+``elasticsearch.scroll-size``           Maximum number of hits to be returned with each batch of Elasticsearch scroll
+``elasticsearch.scroll-timeout``        Number of milliseconds Elasticsearch would keep live for a search scroll result
+``elasticsearch.max-hits``              Maximum number of hits a single Elasticsearch request could fetch
+``elasticsearch.request-timeout``       Timeout for Elasticsearch requests
+======================================= ==============================================================================
+
+``elasticsearch.table-names``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Comma-separated list of all tables provided by this catalog. A table name
+can be unqualified (simple name) and will be put into the default schema
+(see below) or qualified with a schema name (``<schema-name>.<table-name>``).
+
+For each table defined here, a table description file must exist (see below).
+
+This property is required; there is no default and at least one table must be defined.
+
+``elasticsearch.default-schema``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Defines the schema that will contain all the tables defined without
+a qualifying schema name.
+
+This property is optional; the default is ``default``.
+
+``elasticsearch.table-description-dir``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Specifies a path under the Presto deployment directory that contains
+one or more JSON files with table descriptions (must end with ``.json``).
+
+This property is optional; the default is ``etc/elasticsearch``.
+
+``elasticsearch.scroll-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines the maximum number of hits to be returned with each
+batch of Elasticsearch scroll.
+
+This property is optional; the default is ``1000``.
+
+``elasticsearch.scroll-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+This property defines number of milliseconds Elasticsearch client would
+keep live for a search scroll result.
+
+This property is optional; the default is ``20s``.
+
+``elasticsearch.max-hits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines the maximum number of hits an Elasticsearch request can fetch.
+
+This property is optional; the default is ``1000000``.
+
+``elasticsearch.request-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines timeout for all Elasticsearch request.
+
+This property is optional; the default is ``10s``.
+
+Table Definition Files
+----------------------
+
+For Presto, Elasticsearch data must be mapped to tables and columns to allow queries against the data.
```

Comment:
`Elasticsearch data must be mapped to columns/tables for Presto to run queries against them.`